### ISSUE TITLE
Examples with explanations in doc

### ISSUE THREAD

**Issue by Okroshiashvili** [joerick/pyinstrument#233]
It would be really helpful for newcomers to have example outputs and explanations in docs.
For example, I spent few hours looking what `[self]`  means in HTML output. I'm sure there will be other cases when there is something different than `[self]` and it's hard for newcomers to explain why their code spends most of the time there.

**Comment by joerick** [comment]
Yeah, great idea! Are there any other things that tripped you up as a newcomer?

**Comment by Okroshiashvili** [comment]
I guess, yes @joerick 

For example, if a class defines `__call__` method. There will be something... not sure what but...
